### ISSUE TITLE
Clarify plugin-specific keys

### ISSUE THREAD

**Issue by sdavids** [editorconfig/specification#59]
[Supported Pairs](https://spec.editorconfig.org/#supported-pairs)

> Here is the list of all keys defined by this version of this specification

[IntelliJ](https://www.jetbrains.com/help/idea/editorconfig.html#heqdcl_31)

> Generic IntelliJ options that have the `ij_` prefix and are applicable to all languages

[Visual Studio](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/code-style-rule-options#example-editorconfig-file)

> dotnet_separate_import_directive_groups = false
> csharp_preserve_single_line_blocks = true
---

Please clarify how plugins/editors should extend the [existing keys](https://spec.editorconfig.org/#supported-pairs).

One "MUST", "SHOULD", or "MAY" add a prefix/suffix? — [Key words for use in RFCs to Indicate Requirement Levels](https://www.rfc-editor.org/rfc/rfc2119)

How would you add a new core property key in the future without requiring them?

**Comment by xuhdev** [comment]
I agree we should clarify this. Technically, we haven't settled on what to do in this case. I'm thinking that maybe we can require future customized keys to start with `_editor_string_`. Thoughts?

**Comment by ffes** [comment]
I understand and like your idea to specify that custom keys should start with an underscore, but there are already quite a few editors that use a prefix without an underscore.

I don't think there is much we can do about that. Those large companies are unlikely to change their implementions.

I feel that best we can do is to formalize what is already being used right now.

**Comment by xuhdev** [comment]
We can exempt what's already there and encourage transition though.

**Comment by cxw42** [comment]
I think we should move this issue to editorconfig/editorconfig for wider visibility.  This issue seems to be about establishing a new practice, not about documenting an existing guideline.
